lib/pipeline.js

```diff
@@ -743,11 +743,13 @@ class PipelineModel extends BaseModel {
      * Update or create a pipeline given a scmUrl
      * @method _createOrUpdatePipeline
      * @param {String} scmUrl  checkout url for a repository
+     * @param {String} newState  new state for the pipeline associated with the specified scmUrl
      * @return {Promise}
      */
-    async _createOrUpdatePipeline(scmUrl) {
+    async _createOrUpdatePipeline(scmUrl, newState) {
         const { admins } = this;
         const newAdmins = admins;
+        const shouldCreateOrSyncPipeline = newState !== 'INACTIVE';
```

Comment:
let's name it on what the action is, eg: `shouldCreatePipeline` vs `shouldSyncPipeline`

Reply:
Simplified the logic per discussion